src/Common/Infrastructure/Controllers/ApiController.cs

```diff
@@ -156,7 +156,11 @@ public abstract class ApiController<T> : PublicApiController
                     response.HeadersToKeyValuePairs(),
                     response.StatusCode);
             }
-            if (response.StatusCode is HttpStatusCode.OK or HttpStatusCode.Created or HttpStatusCode.Accepted)
+            if (response.StatusCode is HttpStatusCode.OK
+                or HttpStatusCode.Created
+                or HttpStatusCode.Accepted
+                or HttpStatusCode.PreconditionFailed
```

Comment:
I had to add PreconditionFailed & NoContent to be able to get a correct response.

Reply:
`PreconditionFailed` is not an OK error code.
Checkout `HandleBadRequest` in the `RegistryApiController`.

Reply:
This piece of code simply returns what it has received, HandleBadRequest throws an ApiException which has a completely different problem detail, is this the behaviour we expect from public api?